test/acceptance/features/steps/steps.py

```diff
@@ -502,8 +520,41 @@ def validate_secret_empty(context, secret_name):
         pass
 
 
+@then(u'The application got redeployed {count} times so far')
+def check_generation(context, count):
+    context.latest_application_generation = context.application.get_generation()
+    assert context.latest_application_generation - context.original_application_generation == int(count), "Unexpected number of application redeployments"
+
+
+@then(u'The application does not get redeployed again with {time} minutes')
+def check_no_redeployment(context, time):
+    try:
+        polling2.poll(lambda: context.application.get_generation() > context.latest_application_generation, step=5, timeout=int(time)*60)
+        assert False, "Application has redeployed again unexpectedly"
+    except polling2.TimeoutException:
+        pass
+
+
 @given(u'"{app_name}" is deployed from image "{image_ref}"')
 def create_deployment(context, app_name, image_ref):
     app = App(app_name, context.namespace.name, image_ref)
     if not app.is_running():
         assert app.install() is True, "Failed to create deployment."
+
+
+@then(u'Secret of Service Binding "{sb_name}" should be updated')
+def validate_secret_name(context, sb_name):
+    openshift = Openshift()
+    json_path = "{.status.secret}"
+
+    output = polling2.poll(lambda:
+            openshift.get_resource_info_by_jsonpath("servicebindings", sb_name, context.namespace.name, json_path), step=5, timeout=800,
+                ignore_exceptions=(ValueError,))
+    assert output is not None, f"Invalid result for SBO jsonpath: {output}."
```

Comment:
we should do polling until get result different from none.

Reply:
e7384ea